### DISCUSSION TITLE
Issues with VEP Script

### DISCUSSION THREAD

**Discussion by youssefb99** [songlab-cal/gpn#18] [Q&A] [answered]
Title:  Using Hugging Face Transformers

Hello,

I hope this post finds you well. I'm currently facing challenges running the VEP inference script. I've faced an error related to mixed precision training when attempting to run the script on GPU. As an alternative, I'm interested in exploring the possibility of running the code on CPU instead.

![image](https://github.com/songlab-cal/gpn/assets/124166515/fffdac3f-79ca-4ce0-88a4-de4b297d810b)

This is the error I get:
![image](https://github.com/songlab-cal/gpn/assets/124166515/31fcc759-2bf7-4f06-a012-7ad03f0eb38d)
Namespace(center_window_size=None, command='vep', dataloader_num_workers=0, disable_aux_features=False, input_path='example.vcf', is_file=True, model_path='songlab/gpn-msa-sapiens', msa_path='zip:///::https://huggingface.co/datasets/songlab/multiz100way/resolve/main/89.zarr.zip', output_path='example.preds.parquet', per_device_batch_size=25, split='test', window_size=128)
Loading MSA...
Loading MSA... Done

ValueError: FP16 Mixed precision training with AMP or APEX (`--fp16`) and FP16 half precision evaluation (`--fp16_full_eval`) can only be used on CUDA or NPU devices or certain XPU devices (with IPEX).

torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
gpn.msa.inference FAILED

I've successfully run the simple_example script, indicating that the library is working well in my environment.
However, when attempting to use the vep script, I encountered the above error related to mixed precision training.
Is it possible to run the VEP inference script on CPU instead of GPU?
If yes, what adjustments should I make to the code or configurations to facilitate running on CPU?

Thank you in advance for any guidance or insights you can provide!

**Comment by gonzalobenegas** [comment, marked as answer]
Hi, sorry for your troubles! fp16 should certainly be an optional argument. For now, you can remove this line or set to False:
https://github.com/songlab-cal/gpn/blob/05b23c54c572813810c094b31031901f7109575b/gpn/msa/inference.py#L39
Additionally, in the notebook you might need to replace
`{torchrun_path} --nproc_per_node={n_gpu}` with `{python_path}` (which could just be `python`)
Let me know if you get it to work.

**Comment by youssefb99** [comment]
Hello gonzalo,
Thanks for your prompt answer.
I made the necessary adjustments and the code now works just fine.
Thanks again for your help.
